### DISCUSSION TITLE
Ontoportal upgrade and data migration

### DISCUSSION THREAD

**Discussion by gturrisi-lifewatch** [ontoportal/ontoportal-project#3] [Q&A] [answered]
Hello everybody.
I am looking for information about data migration from version 2.5 to 3.1.1. but I'm not able to find information online. Are there any guidelines or documentation that can be helpful for data migration? 
Thank you

**Comment by syphax-bouazzouni** [comment]
Hi @gturrisi-lifewatch, 

I think @alexskr and @mdorf  are working on it currently (in their case is to do the migration from 4store to AllegroGraph), they have done this documentation, see [the Metadata migration section](https://ontoportal.github.io/administration/management/reference_triple_store_rebuild/#metadata-migration). the documentation is under construction and is not really clear  (in my opinion) for now.

In summary, what you need to do is 

### In your current server/instance
1. Dump 4store
```sh
sudo 4s-dump http://localhost:8081/sparql/ -a -o <path where to create the dump>
```
3. Save the repository folder  
```sh
sudo cp /srv/ncbo/repository <path where you  want to save it>
```

### In your new server/instance

> you will certainly use [scp (ssh copy) ](http://www.delafond.org/traducmanfr/man/man1/scp.1.html) to copy the dump folder + repository folder from the old server to the new one. 

1. Stop all the services
```sh
sudo opstop
```
2. Copy the repository folder
```sh
sudo cp <path where you saved your repository folder> /srv/ontoportal/data/repository 
```
3. Restore 4store
```sh
cd <path to your dump directory> && find . -type f | 4s-restore ontologies_api
```

4. Start all the services
```sh
sudo opstop
```
5. Reprocessing all ontologies
```sh
cd /srv/ontoportal/ncbo_cron
bundle exec bin/ncbo_ontology_process -a 
bundle exec bin/ncbo_ontology_metrics -a
```

**Reply by gturrisi-lifewatch** [reply]
Hi @syphax-bouazzouni, 
thanks for your answer and for the indications. 
In the next few days I will do an export / import test in a development environment. 
Thanks again

**Comment by graybeal** [comment]
Hi @gturrisi-lifewatch, @alexskr  is out today but he has a good sense of which versions require what kind of migration. I would definitely encourage talking to him before trying your migration. I am pretty sure there is an explicit document addressing your use case, but I'm not sure if it's in GitHub, a manual, or revision history information for each release (that's where I'd start first).

**Reply by gturrisi-lifewatch** [reply]
Hi @graybeal, many thanks for the suggestion. 
I will wait until @alexskr will come back to ask him for information about the migration, in the form of manuals, guidelines or simply advice.

**Comment by alexskr** [comment, marked as answer]
Hi @gturrisi-lifewatch,
Instructions on how to update v2.5 to v3 appliance can be found here:
  https://ontoportal.github.io/administration/management/appliance_upgrade_v25_to_31   (edited to reflect rename)
Please let me know if you run into any issues with the migration.

**Comment by graybeal** [comment]
@alexskr can you please update that page to provide at least a comment, or perhaps a section, on upgrading to 3.1? For example, we could say "to 3.0 or 3.1" in the title if the instructions are the same. And instructions to go from 3.0 to 3.1 may be obvious to some, but not others, so they would be helpful.

**Comment by alexskr** [comment]
I should have called it an upgrade from 2.5 to 3.1 in the first place.   I updated file name to reflect that change
https://ontoportal.github.io/administration/management/appliance_upgrade_v25_to_31
updating 3.0 -> 3.1 is described here
[here](https://github.com/ontoportal/administration/blob/master/docs/_management/appliance_upgrade.md#v30x-to-v31x-update)

**Comment by graybeal** [comment]
I cross-linked the two pages containing ontoportal upgrade information, https://ontoportal.github.io/administration/management/appliance_upgrade_v25_to_31 and https://ontoportal.github.io/administration/management/appliance_upgrade 

Someone is welcome to make a combined page that is a bit easier to digest.

**Comment by gturrisi-lifewatch** [comment]
Thank you so much @graybeal and @alexskr. 
The instructions for upgrading virtual appliance from v2.5 to v3.1 seem very clear and detailed. In the next few days I will set up a local test environment to perform the migration procedure, before bringing it to our production environment.
If you deem it appropriate, I can create a merge of the 2 pages in order to have all the upgrade procedures in one page: "https://ontoportal.github.io/administration/management/appliance_upgrade/".
Thank you

**Comment by graybeal** [comment]
Giuseppe,

We would welcome your pull request for that! Thanks.

John

On Nov 10, 2022, at 7:22 AM, Giuseppe Turrisi ***@***.******@***.***>> wrote:



Thank you so much @graybeal<https://github.com/graybeal> and @alexskr<https://github.com/alexskr>.
The instructions for upgrading virtual appliance from v2.5 to v3.1 seem very clear and detailed. In the next few days I will set up a local test environment to perform the migration procedure, before bringing it to our production environment.
If you deem it appropriate, I can create a merge of the 2 pages in order to have all the upgrade procedures in one page: "https://ontoportal.github.io/administration/management/appliance_upgrade/".
Thank you

—
Reply to this email directly, view it on GitHub<https://github.com/ontoportal/ontoportal-project/discussions/3#discussioncomment-4108607>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAJVJUBJ6REET6FSOMC56GTWHUHLZANCNFSM6AAAAAARZEISLE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

========================
John Graybeal
Technical Program Manager
Center for Expanded Data Annotation and Retrieval /+/ NCBO BioPortal
Stanford Center for Biomedical Informatics Research
650-736-1632  | ORCID  0000-0001-6875-5360
